x-pack/plugins/ml/public/privilege/get_privileges.ts

```diff
@@ -89,40 +107,59 @@ export function getPrivileges() {
   });
 }
 
-function setGettingPrivileges(cluster = {}, privileges = {}, forceTrue = false) {
+function setGettingPrivileges(
+  cluster: Cluster = {},
+  privileges: Privileges = {},
+  forceTrue = false
+) {
+  // Anomaly Detection
   if (
     forceTrue ||
     (cluster['cluster:monitor/xpack/ml/job/get'] &&
-    cluster['cluster:monitor/xpack/ml/job/stats/get'])
+      cluster['cluster:monitor/xpack/ml/job/stats/get'])
   ) {
     privileges.canGetJobs = true;
   }
 
   if (
     forceTrue ||
     (cluster['cluster:monitor/xpack/ml/datafeeds/get'] &&
-    cluster['cluster:monitor/xpack/ml/datafeeds/stats/get'])
+      cluster['cluster:monitor/xpack/ml/datafeeds/stats/get'])
   ) {
     privileges.canGetDatafeeds = true;
   }
 
+  // Calendars
   if (forceTrue || cluster['cluster:monitor/xpack/ml/calendars/get']) {
     privileges.canGetCalendars = true;
   }
 
+  // Filters
   if (forceTrue || cluster['cluster:admin/xpack/ml/filters/get']) {
     privileges.canGetFilters = true;
   }
 
+  // File Data Visualizer
   if (forceTrue || cluster['cluster:monitor/xpack/ml/findfilestructure']) {
     privileges.canFindFileStructure = true;
   }
+
+  // Data Frames
+  if (
+    forceTrue ||
+    (cluster['cluster:monitor/data_frame/get'] && cluster['cluster:monitor/data_frame/stats/get'])
+  ) {
+    privileges.canGetDataFrameJobs = true;
+  }
 }
 
-function setActionPrivileges(cluster = {}, privileges = {}) {
-  if (cluster['cluster:admin/xpack/ml/job/put'] &&
+function setActionPrivileges(cluster: Cluster = {}, privileges: Privileges = {}) {
+  // Anomaly Detection
+  if (
+    cluster['cluster:admin/xpack/ml/job/put'] &&
     cluster['cluster:admin/xpack/ml/job/open'] &&
-    cluster['cluster:admin/xpack/ml/datafeeds/put']) {
+    cluster['cluster:admin/xpack/ml/datafeeds/put']
+  ) {
     privileges.canCreateJob = true;
```

Comment:
outside of this scope, but these settings should probably all be renamed to `canCreateAnomalyDetectionJob` at some point.